nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -372,13 +372,12 @@ in
       preStart =  mkOption {
         type = types.lines;
         default = ''
-          test -d ${cfg.stateDir}/logs || mkdir -m 750 -p ${cfg.stateDir}/logs  
+          test -d ${cfg.stateDir}/logs || mkdir -p -m 750 ${cfg.stateDir}/logs
           test `stat -c %a ${cfg.stateDir}` = "750" || chmod 750 ${cfg.stateDir}
           test `stat -c %a ${cfg.stateDir}/logs` = "750" || chmod 750 ${cfg.stateDir}/logs
-          chown -R ${cfg.user}:${cfg.group} ${cfg.stateDir}
         '';
         description = "
-          Shell commands executed before the service's nginx is started.
+          Shell commands executed as root before the service's nginx is started.
```

Comment:
This code won't be run as root since you've removed PermissionsStartOnly.

Reply:
A solution would be:

```
systemd.services.nginx.serviceConfig.ExecStartPre = "+" + pkgs.writeScript "nginx-prestart" ''
  ${cfg.preStart}
  ${cfg.package}/bin/nginx -c ${configFile} -p ${cfg.stateDir} -t
'';
```

Reply:
Sorry, remove update description. Not need root access.